cronjobs/nb-culler/cronjob.yaml

```diff
@@ -0,0 +1,81 @@
+kind: CronJob
+apiVersion: batch/v1
+metadata:
+  name: nb-culler
+  labels:
+    component.opendatahub.io/name: nb-culler
+    opendatahub.io/component: 'true'
+    opendatahub.io/modified: 'false'
+spec:
+  schedule: '0 * * * *'
+  startingDeadlineSeconds: 200
+  concurrencyPolicy: Replace
+  suspend: false
+  jobTemplate:
+    metadata:
+      labels:
+        component.opendatahub.io/name: nb-culler
+        opendatahub.io/component: 'true'
+    spec:
+      template:
+        metadata:
+          labels:
+            component.opendatahub.io/name: nb-culler
+            opendatahub.io/component: 'true'
+            parent: nb-culler
+        spec:
+          restartPolicy: Never
+          serviceAccountName: nb-culler
+          schedulerName: default-scheduler
+          terminationGracePeriodSeconds: 30
+          securityContext: {}
+          containers:
+            - name: oc-cli
+              image: >-
+                registry.redhat.io/openshift4/ose-cli@sha256:25fef269ac6e7491cb8340119a9b473acbeb53bc6970ad029fdaae59c3d0ca61
+              command: ["/bin/bash", "-c", "--"]
+              args:
+              - |
+                  group_name="ee440"
+                  notebooks=$(oc get notebooks -n ope-rhods-testing-1fef2f -o jsonpath="{range .items[?(@.status.containerState.running)]}{.metadata.name}{' '}{.metadata.namespace}{' '}{.status.containerState.running.startedAt}{' '}{.metadata.annotations['opendatahub\.io/username']}{' '}{.metadata.annotations['notebooks\.opendatahub\.io/last-image-selection']}{' '}{.metadata.annotations['notebooks\.opendatahub\.io/last-size-selection']}{'\n'}{end}")
+                  if [ -z "$notebooks" ]; then
+                      echo "No running notebooks found"
+                      exit 0
+                  fi
+                  group_members=$(oc get group $group_name -o=jsonpath='{.users[*]}')
+
+                  # Loop through each notebook
+                  while read -r nb ns ts user image size; do
+                      if [[ " $group_members " =~ " $user " ]]; then
+                          echo "$user is in the $group_name group."
+                          cutoff_time=21600
+                          current_time=$(date -u +%s)
+                          timestamp=$(date -d $ts +%s)
+                          difference=$((current_time - timestamp))
+                          if [[ $image != *"ucsls-nerc-rhoai"* ]]; then
```

Comment:
Let's maybe make this configurable via variable (The image we're checking against) as well? Same reason as  suggestion above: [https://github.com/nerc-project/operations/issues/404](https://github.com/nerc-project/operations/issues/404)

Reply:
I've set these 3 variables to ENV variables and tested them. It's working now.